stdlib/filename.mli

```diff
@@ -183,9 +183,21 @@ val get_temp_dir_name : unit -> string
 (** The name of the temporary directory:
     Under Unix, the value of the [TMPDIR] environment variable, or "/tmp"
     if the variable is not set.
-    Under Windows, the value of the [TEMP] environment variable, or "."
-    if the variable is not set.
+
+    Under Windows, it uses [GetTempPath2] (since Windows 10 21H1), or
+    [GetTempPath], and returns:
+    - if the process runs as SYSTEM, the path "C:\\Windows\\SystemTemp\\"
+      (since Windows 10 21H1); or
+    - the path specified by the [TMP] environment variable; or
+    - the path specified by the [TEMP] environment variable; or
+    - the path specified by the [USERPROFILE] environment variable; or
+    - the Windows directory.
+
     The temporary directory can be changed with {!Filename.set_temp_dir_name}.
+
+    Under Windows, before OCaml 5.4, it would return the value of the
+    [TEMP] environment variable, or "." if the variable was not set.
```

Comment:
If I understand correctly, the new function never returns `.` anymore. This is a minor break in backwards compatibility, right?

Reply:
Yes, and the changelog entry lists a breaking change.
Should I make that clearer?